numba/core/event.py

```diff
@@ -465,6 +465,13 @@ def _prepare_chrome_trace_data(listener: RecordingListener):
     return evs
 
 
+class _LazyJSONEncoder(json.JSONEncoder):
```

Comment:
Might be slightly safer/cleaner to move this to `utils.py` alongside `_lazy_pformat` so any other location that later needs the json encoder for `utils._lazy_pformat` can easily find it. I understand this is the only usage but it seems like it would decrease the likelihood of duplicated code.